File: Easy/[231] Power of Two/Python3 Solution/[231] Power of Two.py
"""
Accepted
231 [Easy]
Runtime: 37 ms, faster than 55.59% of Python3 online submissions for Power of Two.
Memory Usage: 16.60 MB, less than 53.28% of Python3 online submissions for Power of Two.
"""
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and (2**30 % n) == 0
    


"""
Accepted
231 [Easy]
Runtime: 30 ms, faster than 90.28% of Python3 online submissions for Power of Two.
Memory Usage: 16.50 MB, less than 96.02% of Python3 online submissions for Power of Two.
"""
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and (n & (n - 1)) == 0
    


"""
Accepted
231 [Easy]
Runtime: 27 ms, faster than 96.66% of Python3 online submissions for Power of Two.
Memory Usage: 16.53 MB, less than 74.80% of Python3 online submissions for Power of Two.
"""
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n > 0 and math.pow(2, int(math.log2(n))) == n:
            return True
        else:
            return False

File: Easy/[231] Power of Two/Java Solution/[231] Power of Two.java
/*
Accepted
231 [Easy]
Runtime: 2 ms, faster than 11.15% of Java online submissions for Power of Two.
Memory Usage: 40.87 MB, less than 43.43% of Java online submissions for Power of Two.
 */
class Solution {
    public boolean isPowerOfTwo(int n) {
        int flag=0,fl=0;
        if(n<0)
            return false;
        if(n==1)
            flag=1;
        while(n!=0) {
            if(n%2!=0 && n!=1)
                return false;
            if(n==2)
                fl=1;
            n=n/2;
        }
        if(flag==1 || fl==1)
            return true;
        else
            return false;
    }
}